src/core/vector/qgsvectorlayertemporalproperties.cpp

```diff
@@ -427,24 +433,27 @@ QString QgsVectorLayerTemporalProperties::createFilterString( const QgsVectorLay
       if ( mAccumulateFeatures )
       {
         return QStringLiteral( "(%1 %2 %3) OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
-               range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
-               dateTimeExpressionLiteral( range.end() ) );
+               vectorRange.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
+               dateTimeExpressionLiteral( vectorRange.end() ) );
       }
       else if ( qgsDoubleNear( mFixedDuration, 0.0 ) )
       {
+        // Working with features with a momentarily duration (so only one timestamp and duration about zero)
+        // You really need to take at least ONE limit into account (in practice mostly the beginning)
         return QStringLiteral( "(%1 %2 %3 AND %1 %4 %5) OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
-               range.includeBeginning() ? QStringLiteral( ">=" ) : QStringLiteral( ">" ),
-               dateTimeExpressionLiteral( range.begin() ),
-               range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
-               dateTimeExpressionLiteral( range.end() ) );
+               ">=",  // always taking the begin into account
+               dateTimeExpressionLiteral( vectorRange.begin() ),
+               vectorRange.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
+               dateTimeExpressionLiteral( vectorRange.end() ) );
       }
       else
       {
-        // Working with features with events with a duration, so taking this duration into account (+ QgsInterval( -mFixedDuration, mDurationUnit ) ))
-        // Then we are NOT taking the range.includeBeginning() and range.includeEnd() into account (deliberately, see #38468)
-        return QStringLiteral( "(%1 > %2 AND %1 < %3) OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
-               dateTimeExpressionLiteral( range.begin() + QgsInterval( -mFixedDuration, mDurationUnit ) ),
-               dateTimeExpressionLiteral( range.end() ) );
+        // Working with features with events with a duration
+        return QStringLiteral( "(%1 %2 %3 AND %1 %4 %5) OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
+               vectorRange.includeBeginning() ? QStringLiteral( ">=" ) : QStringLiteral( ">" ),
```

Comment:
Yep, and as above, here the story is the other way around:
- having both a view range with a duration AND features with a time and a duration you NEVER want to add limits into your filter range, because that would mean that your features will fall into 2 range classes ( see https://github.com/qgis/QGIS/issues/38468 )